Sources/PackageModelSyntax/AddPackageDependency.swift

```diff
@@ -55,6 +62,52 @@ public enum AddPackageDependency {
         )
     }
 
+    /// Check that the given package dependency doesn't already exist in the manifest.
+    /// If the exact same dependency already exists, `false` is returned to indicate
+    /// that no edits are needed. If a different dependency with the same id or url
+    /// with different arguments exists, an error is thrown.
+    private static func checkExistingDependency(
```

Comment:
question: I know this is marked as a private function, but is there value is validate this function in isolation to ensure we can test all code paths to ensure the behaviour is as expected, as opposed to testing the implementation?